wimm.Secundatives/Extensions/FlattenExtensions.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System.Threading.Tasks;
+
+namespace wimm.Secundatives.Extensions
+{
+    /// <summary>
+    /// Extensions for converting nested, compatible monads into un-nested monads
+    /// </summary>
+    public static class FlattenExtensions
+    {
+        /// <summary>
+        /// Un-nest a nested result.
+        /// </summary>
+        /// <typeparam name="TValue"> The value type of the nested <see cref="Result{T, TError}"/></typeparam>
+        /// <typeparam name="TErr"> The error type of the nested <see cref="Result{T, TError}"/></typeparam>
+        /// <param name="result"> The nested result to be unwrapped and converted to <see cref="Result{TValue, TError}"/></param>
+        /// <returns> 
+        /// The un-nested result of inspecting the outer result for error and returning it if it exists or the nested result itself otherwise 
+        /// </returns>
+        public static Result<TValue, TErr> Flatten<TValue, TErr>(this Result<Result<TValue, TErr>, TErr> result)
+        {
+            return result.IsValue ? result.Value : result.Error;
```

Comment:
Should one call to `Flatten` flatten all the way to the first non-`Result` type? I guess we'd lose type safety since we'd have to pun the first child type though 🤔 This language sucks you should switch to `Go`

Reply:
Feel shame. 

We could do any number with reflection but unless that's an emergent edge case I don't wanna dip into that unless we have to. Might be able to sneak a recursive case around the compiler but it would be wonky.

Reply:
Basically. I think we could do it but it'd be a bunch of reflection and I'd prefer to see if that case comes up often enough for the perf/wackiness.